TileTowns/Assets/Scripts/Managers/RoadManager.cs

```diff
@@ -0,0 +1,189 @@
+    }
+
+    private void CreateRoad(ITileData tileData)
+    {
+        var position = tileData.Position;
+        _roadMap[position] = true;
+        
+        var adjacentTiles = GetAdjacentPositions(position);
+        foreach (var adjacentPosition in adjacentTiles)
+        {
+            if (_roadMap.ContainsKey(adjacentPosition) && _roadMap[adjacentPosition])
+            {
+                UpdateRoad(adjacentPosition);
+            }
+        }
+        
+        UpdateRoad(position);
+    }
+
+    private void UpdateRoad(Vector2Int position)
+    {
+        bool hasRoadAbove = false;
+        bool hasRoadBelow = false;
+        bool hasRoadLeft = false;
+        bool hasRoadRight = false;
+
+        if (_roadMap[position + new Vector2Int(0, 1)])
+        {
+            hasRoadAbove = true;
+        }
+
+        if (_roadMap[position + new Vector2Int(0, -1)])
+        {
+            hasRoadBelow = true;
+        }
+
+        if (_roadMap[position + new Vector2Int(1, 0)])
+        {
+            hasRoadRight = true;
+        }
+
+        if (_roadMap[ position + new Vector2Int(-1, 0)])
+        {
+            hasRoadLeft = true;
+        }
+        
+        if (hasRoadAbove && hasRoadBelow && hasRoadLeft && hasRoadRight)
+        {
+            SetRoad(position, upDownLeftRightRoad);
+        }
+        else if (hasRoadAbove && hasRoadBelow && hasRoadLeft)
+        {
+            SetRoad(position, upDownLeftRoad);
+        }
+        else if (hasRoadAbove && hasRoadBelow && hasRoadRight)
+        {
+            SetRoad(position, upDownRightRoad);
+        }
+        else if (hasRoadAbove && hasRoadLeft && hasRoadRight)
+        {
+            SetRoad(position, upLeftRightRoad);
+        }
+        else if (hasRoadBelow && hasRoadLeft && hasRoadRight)
+        {
+            SetRoad(position ,downLeftRightRoad);
+        }
+        else if (hasRoadAbove && hasRoadBelow)
+        {
+            SetRoad(position, upDownRoad);
+        }
+        else if (hasRoadAbove && hasRoadLeft)
+        {
+            SetRoad(position, upLeftRoad);
+        }
+        else if (hasRoadAbove && hasRoadRight)
+        {
+            SetRoad(position, upRightRoad);
+        }
+        else if (hasRoadBelow && hasRoadLeft)
+        {
+            SetRoad(position, downLeftRoad);
+        }
+        else if (hasRoadBelow && hasRoadRight)
+        {
+            SetRoad(position, downRightRoad);
+        }
+        else if (hasRoadLeft && hasRoadRight)
+        {
+            SetRoad(position, leftRightRoad);
+        }
+        else if (hasRoadAbove)
+        {
+            SetRoad(position, upRoad);
+        }
+        else if (hasRoadBelow)
+        {
+            SetRoad(position, downRoad);
+        }
+        else if (hasRoadLeft)
+        {
+            SetRoad(position, leftRoad);
+        }
+        else if (hasRoadRight)
+        {
+            SetRoad(position, rightRoad);
+        }
+        else
+        {
+            // No roads around
```

Comment:
you can add InvalidOperationException or something (optional)

Reply:
It's just a scenario, where it'a a lone building, don't think it should throw any exception

Reply:
ow wait fuck xD
didnt read into it, you can write the `intentionally empty`, thats quite a good practice but however you prefer ;)